### PR TITLE
Use prettier-plugin-hermes-parser to match upstream settings

### PR DESCRIPTION
**Pull request by rozele**
### What
In order to minimize the delta between platform override files in react-native-windows and upstream facebook/react-native files, switch to prettier-plugin-hermes-parser. The same change was made in https://github.com/facebook/react-native/commit/dc2037c0f1c06fe2a176ea48c6f255d10fa9e4b7

## Changelog
Should this change be included in the release notes: _no_
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12599)

### CODE REVIEW

File: vnext/src/Libraries/LogBox/UI/LogBoxInspectorHeader.windows.js
Comment (jonthysell) [inline vnext/src/Libraries/LogBox/UI/LogBoxInspectorHeader.windows.js:91]: It looks like you need to apply this change to `packages\@office-iss\react-native-win32\src\Libraries\LogBox\UI\LogBoxInspectorHeader.win32.js` in order to pass linting.

**Review by jonthysell — APPROVED** [review]

**Review by marlenecota — APPROVED** [review]

**Comment by chiaramooney** [comment]
Change is approved. Will merge when CI runs green.

**Comment by chrisglein** [comment]
> Change is approved. Will merge when CI runs green.

@rozele PR has signoff but needs a clean CI run. With that, good to go.

**Review by marlenecota — APPROVED** [review]

**Pull request merged**
